client/src/views/page-components/MainPage/CategorySlide/CategorySlide.js

```diff
@@ -12,11 +18,50 @@ export default function CategorySlide({ $selector }) {
     return this.$CategorySlide.classList.add('slide-trigerred');
   };
 
-  this.render = () => {
+  // state
+
+  this.setState = (nextState) => {
+    this.state = nextState;
+    this.checkCurrentCategory(this.state.currentCategory);
+  };
+
+  this.checkCurrentCategory = (currentCategory) => {
+    this.render(currentCategory);
+  };
```

Comment:
아래 처럼 구현할 수 있을 것 같은데, 
혹시 코드의 가독성을 위해서 함수를 하나 더 작성한 것일까요??
```suggestion
  this.setState = (nextState) => {
    this.state = nextState;
    this.render(this.state.currentCategory)
  };
```

Reply:
네 맞습니다 흠 근데 별로 가독성이 좋아보이지 않네요 수정하겠습니다 🤣